### PR TITLE
Added XDC Network MainNet & TestNet in TradeTrust Ecosystem

### PR DESCRIPTION
**Pull request by AnilChinchawale**
## Summary

Add support for XDC Network & Apothem TestNet

## Changes

We have added Apothem and Mainnet XDC network configurations.

image

### CODE REVIEW

File: src/commands/wallet/create.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export const builder = (yargs: Argv): Argv =>
     .option("fund", {
       description: "Automatically add funds for the specified network",
       type: "string",
-      choices: ["ropsten"],
+      choices: ["ropsten","apothem"],
```
Comment (superical) [inline src/commands/wallet/create.ts:25]: Fund from where?

File: src/commands/networks.ts [hunk 2]
```diff
@@ -12,6 +12,8 @@ enum NetworkCmdName {
   Rinkeby = "rinkeby",
   Goerli = "goerli",
   Sepolia = "sepolia",
+  XDC = "xdc",
+  Apothem = "apothem",
```
Comment (superical) [inline src/commands/networks.ts:16]: Move them below Mumbai.
https://github.com/Open-Attestation/open-attestation-cli/blob/7601147461e89b12608a5e06b5d67cb10f2c5746/src/commands/networks.ts#L17-L18

File: src/implementations/wallet/create.ts [hunk 3]
```diff
@@ -36,6 +36,19 @@ export const create = async ({
       );
     }
   }
+  else if (fund === "apothem") {
+    const response = await fetch(`https://faucet.openattestation.com/donate/${wallet.address}`).then((res) =>
```
Comment (superical) [inline src/implementations/wallet/create.ts:40]: OpenAttestation faucet doesn't support Apothem, does it?

**Review by superical — COMMENTED** [review]
There are a few other places and repos you need to update to support the new network too. I suggest that you also run through a test on your own first to verify that all the commands in the CLI is working with the XDC network.

File: src/implementations/wallet/create.ts (hunk 3 above)
Comment (AnilChinchawale) [reply src/implementations/wallet/create.ts]: XDC Faucet link updated here :- https://faucet.apothem.network

**Comment by AnilChinchawale** [comment]
Can you share the exact command to test the application, master branch yarn test return below output.

Test Suites: 20 failed, 15 passed, 35 total
Tests:       96 failed, 3 todo, 61 passed, 160 total
Snapshots:   3 passed, 3 total
Time:        5.627 s

**Comment by superical** [comment]
> Can you share the exact command to test the application, master branch yarn test return below output.
> 
> Test Suites: 20 failed, 15 passed, 35 total Tests: 96 failed, 3 todo, 61 passed, 160 total Snapshots: 3 passed, 3 total Time: 5.627 s

Try running `npm run test`.

**Comment by AnilChinchawale** [comment]
Here is latest Test Result.

image

**Comment by superical** [comment]
@AnilChinchawale Like the other PR for the website at https://github.com/TradeTrust/tradetrust-website/pull/632#issuecomment-1317766019, my teammate has made changes to the CLI repo to use the latest versions.  The PRs aren't merged yet but you might want to base your changes on his PR first. 

@puayhiang Can help to review this PR so that the changes will align with your PRs? Most importantly to avoid too many change conflicts for you. Changes in your PRs take priority. 🙇‍♂️

**Comment by AnilChinchawale** [comment]
@puayhiang Thank you for your guidance, we will now forward this to @BlocksScanIO Dev Team for further development and will update you once we raised the PR.

**Pull request closed without merging**